runner/internal/shim/api/http.go

```diff
@@ -1,57 +1,53 @@
 package api
 
 import (
-	"encoding/base64"
-	"encoding/json"
+	"context"
+	"fmt"
 	"log"
 	"net/http"
 
-	"github.com/docker/docker/api/types/registry"
 	"github.com/dstackai/dstack/runner/internal/api"
 	"github.com/dstackai/dstack/runner/internal/shim"
 )
 
-func (s *ShimServer) healthcheckGetHandler(w http.ResponseWriter, r *http.Request) (interface{}, error) {
+func (s *ShimServer) HealthcheckGetHandler(w http.ResponseWriter, r *http.Request) (interface{}, error) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
 	return &HealthcheckResponse{
 		Service: "dstack-shim",
+		Version: s.version,
 	}, nil
 }
 
-func (s *ShimServer) registryAuthPostHandler(w http.ResponseWriter, r *http.Request) (interface{}, error) {
+func (s *ShimServer) SubmitPostHandler(w http.ResponseWriter, r *http.Request) (interface{}, error) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
-	if s.state != shim.WaitRegistryAuth {
+	if s.runner.GetState() != shim.Pending {
 		return nil, &api.Error{Status: http.StatusConflict}
 	}
 
-	var body RegistryAuthBody
+	var body DockerTaskBody
 	if err := api.DecodeJSONBody(w, r, &body, true); err != nil {
 		log.Println("Failed to decode submit body", "err", err)
 		return nil, err
 	}
 
-	authConfig := registry.AuthConfig{
-		Username: body.Username,
-		Password: body.Password,
-	}
-	encodedConfig, err := json.Marshal(authConfig)
-	if err != nil {
-		log.Println("Failed to encode auth config", "err", err)
-		return nil, err
-	}
-	s.registryAuth <- base64.URLEncoding.EncodeToString(encodedConfig)
+	go func(taskParams shim.DockerImageConfig) {
```

Comment:
We should have a way to kill a running container without interrupting the instance. I wouldn't rely on the runner in this case.

Reply:
I'm going to make new issue for this.

Reply:
#895